File: 0169-majority-element/0169-majority-element.c
int majorityElement(int* nums, int numsSize) {
    int major=0,c=1,i;
    int n=numsSize;
    for(i=1;i<n;i++)
    {
       if (nums[major]==nums[i]){
            c++;
          }
        else
            c--;
        if (c==0)
        {
            major=i;
            c=1;
        }
    }
    return nums[major];
}

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int n1=nums.length;
        Map<Integer,Integer>map=new HashMap<>();
        for(int num:nums)
        {
            map.put(num,map.getOrDefault(num,0)+1);
        }
        for(Map.Entry<Integer,Integer>en:map.entrySet())
        {
            if(en.getValue()>n1/2)
            {
                return en.getKey();
            }
        }
        return -1;
    }
}